server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -91,7 +91,7 @@ public RemoteFsTranslog(
             Checkpoint checkpoint = readCheckpoint(location);
             this.readers.addAll(recoverFromFiles(checkpoint));
             if (readers.isEmpty()) {
-                throw new IllegalStateException("at least one reader must be recovered");
+                throw new IllegalStateException(shardId + " at least one reader must be recovered");
```

Comment:
For this class, can we use logger defined in `AbstractIndexShardComponent` . It will automatically log the shard level details for all logs automatically .

Reply:
This class has some static methods which use the logger, need to check how to add shard id for those.